.ci/ci-pre-checks.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
+#  This file is part of cc-oci-runtime.
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#---------------------------------------------------------------------
+# Description: Perform basic checks on the branch before attempting
+#   to run the build and test suites. If this script fails, the CI run
+#   should be aborted.
+#---------------------------------------------------------------------
+
+script_name=${0##*/}
+
+# Limits defined in CONTRIBUTING.md
+max_subject_length=75
+max_body_line_length=72
+
+# Log a message to stderr and exit
+die()
+{
+    msg="$*"
+    echo >&2 -e "ERROR: $script_name: commit $commit: $msg"
+    exit 1
+}
+
+# Log an informational string to stdout
+msg()
+{
+    str="$*"
+    echo -e "INFO: $script_name: $str"
+}
+
+# Perform checks on git subject line
+check_commit_subject()
+{
+    local subject="$1"
+
+    [ -n "$subject" ] || die "no subject"
+
+    echo "$subject"|grep -q "^[^ ][^ ]*.*:" \
+        || die "no subsystem in subject"
+
+    len=$(echo -n "$subject"|wc -c)
+
+    [ "$len" -gt "$max_subject_length" ] \
+        && die "subject too long (length $len, max $max_subject_length)"
+}
+
+# Perform checks on git subject line
+check_commit_body()
+{
+    local body="$1"
+
+    local -a lines
+
+    [ -n "$body" ] || die "no body"
+
+    # Load the body text into an array
+    mapfile -t lines < <( echo "$body" )
+
+    local found_fixes=no
+    for line in "${lines[@]}"
+    do
+        echo "$line"|egrep -iq "\<fixes\> *#[0-9][0-9]*"
```

Comment:
Just a note/clarification that we are fixing our requirement here to 'fixes', whereas in fact github has a slew of words that can be used to close a commit: https://help.github.com/articles/closing-issues-via-commit-messages/
So, some acks that this is our intention would be good.